### PR TITLE
ddl: fix the ddl txn commit may be conflict with reorg txn

### PR DESCRIPTION
**Pull request by AilinKid**
Signed-off-by: AilinKid <314806019@qq.com>

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/24427

Problem Summary:

### What is changed and how it works?

What's Changed:
```
// runReorgJob is used to do `READ` action of reorg info in ddl txn.
// If we want to do `WRITE` action of recording some reorg handle down, we need to additional kv txn.
// Otherwise, the below write conflicts will occur.
//
// ddl txn ------------+-------------------------------+
// (RunInKVTxn: start) |                               | (ddl done successfully)
//                     | (Write)                       | (RunInKVTxn committed fail: write conflict)
//                     V                               V
//               reorg handle
//                                    ^
//                                    | (Update handle, eg: change element)
//                                    | (RunInKVTxn: committed instantly)
// reorg txn--------------------------+
//
// For this case:
// Let's take the ddl txn as a Daemon thread, it isn't response for writing the reorg handle down, but care for reading
// reorg result. Because read won't be conflict with any write action of the reorg txn.
```

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

- ddl: fix the ddl txn commit fail cause it's conflict with reorg txn

### CODE REVIEW

File: ddl/reorg.go [hunk 1]
```diff
@@ -220,7 +237,7 @@ func (w *worker) runReorgJob(t *meta.Meta, reorgInfo *reorgInfo, tblInfo *model.
 		case model.ActionModifyColumn:
 			metrics.GetBackfillProgressByLabel(metrics.LblModifyColumn).Set(100)
 		}
-		if err1 := t.RemoveDDLReorgHandle(job, reorgInfo.elements); err1 != nil {
+		if err1 := reorgInfo.CleanReorgMeta(); err1 != nil {
```
Comment (zimulala) [inline ddl/reorg.go:240]: This txn and DDL txn are not the same txn. There may be data and schema out of sync. I think this plan can be further optimized

**Comment by AilinKid** [comment]
This pr is regarded as a kind of improvement. 
For clear, we can clean the element handle in the `finishDDLJob` which is used to clean remained indexes. I will do it latter.

**Comment by tangenta** [comment]
@AilinKid Please resolve the conflicts.

**Comment by AilinKid** [comment]
> @AilinKid Please resolve the conflicts.

got

**Comment by AilinKid** [comment]
won't fix it yet, it will be refactored as a whole latter

**Pull request closed without merging**